_posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md

```diff
@@ -0,0 +1,173 @@
+All these parts are bound to two dates (start date and end date) so the merchant can get even more useful data faster.
+
+{% imagebasic blog/blog-sales-app-dashboard.png %} image {% endimagebasic %}
+
+The main purpose of the Dashboard was to have all the basic information about a shop at a glance.
+So we decided to include a graph of sales, the amount of sales and orders, as well as the top products, and last orders.
+
+Then we included three charts on the Sales section.
+The first is the same as in the Dashboard, the second one represents the orders on the selected time frame and the last is the revenue per order, the calculation between the first two graphs.
+We also added the option to export the data to an `xlsx` file so the merchant doesn't need an internet connection to review the sales.
+
+The Customers section is a simple table including all the merchant customers ordered via number of orders or the total amount spent.
+Above the table we can see all the customers for the timeframe and if we had any returning ones.
+
+The last part are the Products.
+In this view, we have the products ordered via revenue or amount of units sold.
+Here we implemented the functionality of having more insights on the product.
+When clicking on one, you can see the extra data or cross selling products, this is, what other items have shared orders with.
+This allows the merchant to maybe realize on connections (or disconnections) between products in an easy and fast way.
+
+Both, the Customers and Products section, have the option to download the data on a `csv`.
+Other basic functionalities like changing you email address or password and 'Forgot my password' are also included in the app.
+
+## Some decisions we made
+
+In every project you have to make some decisions on what tech you use.
+We decided to use [Rails](http://rubyonrails.org/) - in part because we love it, but also to use our [own developed library](https://github.com/ePages-de/epages-rest-ruby).
+On the charts side there are lots of options to choose, but we went with [D3](https://d3js.org/), an open source, very powerful and very customizable option.
+The main reason of picking this was purely educational.
+I love learning new stuff, libraries, languages, paradigms, you name it; and I already used [Highcharts](http://www.highcharts.com/), [Google charts](https://developers.google.com/chart/) and [Chart.js](http://www.chartjs.org/), plus I was really curious about it and my team let me decide {% emoji smiley %}.
+
+[Devise](https://github.com/plataformatec/devise) was our option for user management.
+This gem is easy and simple, so it was perfect to deliver fast (even though in the end everything became more complicated).
+For this simplicity reason we choose [Slim](http://slim-lang.com/), [Sass](http://sass-lang.com/) and [CoffeeScript](http://coffeescript.org/).
+
+## Issues on the development process
```

Comment:
Start from here for another post.
Need to find a nice headline, image and intro. I can help you out with that.
Once the second one is in place, I'll have another check on it.